src/sonic-yang-models/yang-models/sonic-buffer-profile.yang

```diff
@@ -0,0 +1,67 @@
+module sonic-buffer-profile {
+	namespace "http://github.com/Azure/sonic-buffer-profile";
+	prefix bpf;
+
+	import sonic-buffer-pool {
+		prefix bpl;
+	}
+
+	organization
+		"SONiC";
+
+	contact
+		"SONiC";
+
+	description
+		"SONIC BUFFER PROFILE";
+
+	revision 2019-05-15 {
+		description
+			"Initial revision.";
+	}
+
+
+	container sonic-buffer-profile {
+
+		container BUFFER_PROFILE {
+
+			list BUFFER_PROFILE_LIST {
+				key "name";
+
+				leaf name {
+					type string;
+				}
+
+				leaf static_th {
```

Comment:
We can have either static or dynamic, we may need choice statement here

Reply:
Yes, will add a choice statement here.

Reply:
Praveen, 
   I tried following choice/case statement for the threshold, but got the error mentioned below. 
Statement
=======
                choice threshold_type {
                    mandatory true;
                    case static {
                        leaf static_th {
                            type uint64;
                            description "The maximum size of the buffer pool the port can occupy";
                        }
                    }
                    case dynamic {
                        leaf dynamic_th {
                            type int32 {
                                range "-7..3";
                            }
                            description "The maximum proportion of the free size of the buffer pool the port can occupy (Alpha value)";
                        }
                    }
                }

Error:
====
           self.root = None
           self.sysLog(msg="Data Loading Failed:{}".format(str(e)), \
            debug=syslog.LOG_ERR, doPrint=True)
>          raise SonicYangException("Data Loading Failed\n{}".format(str(e)))
E          sonic_yang_ext.SonicYangException: Data Loading Failed
E          string indices must be integers

Reply:
Praveen, Do you see any issue with the above yang block  for choice.